src/Controls/src/Core/Compatibility/iOS/Extensions/ToolbarItemExtensions.cs

```diff
@@ -26,20 +27,20 @@ public static UIBarButtonItem ToUIBarButtonItem(this ToolbarItem item, bool forc
 		sealed class PrimaryToolbarItem : UIBarButtonItem
 		{
 			readonly bool _forceName;
-			readonly ToolbarItem _item;
+			readonly WeakReference<ToolbarItem> _item;
 
 			public PrimaryToolbarItem(ToolbarItem item, bool forceName)
 			{
 				_forceName = forceName;
-				_item = item;
+				_item = new(item);
 
 				if (item.IconImageSource != null && !item.IconImageSource.IsEmpty && !forceName)
-					UpdateIconAndStyle();
+					UpdateIconAndStyle(item);
 				else
-					UpdateTextAndStyle();
-				UpdateIsEnabled();
+					UpdateTextAndStyle(item);
+				UpdateIsEnabled(item);
 
-				Clicked += (sender, e) => ((IMenuItemController)_item).Activate();
+				Clicked += OnClicked;
```

Comment:
I couldn't change this to:
```suggestion
				Clicked += (sender, e) => ((IMenuItemController)item).Activate();;
```
This created a closure on `item` that also leaked.